lib/puppet/pops/types/type_asserter.rb

```diff
@@ -1,6 +1,7 @@
 # Utility module for type assertion
 #
-module Puppet::Pops::Types::TypeAsserter
+module Puppet::Pops::Types
+module TypeAsserter
```

Comment:
why this odd construct and not just a straight definition of the module? Is it because the Puppet::Pops::Types module does not exist yet and you want it created as an empty module to then redefine?